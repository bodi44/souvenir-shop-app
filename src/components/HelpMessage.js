import React from 'react'

const HelpMessage = () => (
  <div>
    <p>
      <strong>purchase &lt;date&gt; &lt;price&gt; &lt;currency&gt; &lt;productName&gt;</strong> - add purchase to
      store.<br/>
      <i>
        Example: purchase 2019-04-25 12 USD Photo <br/>
        Output: <br/>
        2019-04-25 <br/>
        Photo 12 USD
      </i>
    </p>
    <p>
      <strong>all</strong> - show all purchases. <br/>
      <i>
        Example: all <br/>
        Output: <br/>
        2019-04-25 <br/>
        Photo 12 USD
      </i>
    </p>
    <p>
      <strong>clear &lt;date&gt;</strong> - removes all purchases for specified date. <br/>
      <i>
        Example: clear 2019-04-25 <br/>
        Output:
      </i>
    </p>
    <p>
      <strong>report &lt;year&gt; &lt;currency&gt;</strong> - calculate the total income for specified year,
      convert
      and present it in specified currency. <br/>
      <i>
        Example: report 2019 UAH <br/>
        Output: <br/>
        356.4 UAH
      </i>
    </p>
  </div>
)

export default HelpMessage