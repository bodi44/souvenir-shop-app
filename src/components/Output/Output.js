import React from 'react'
import PropTypes from 'prop-types'

import PurchasesBlock from '../PurchasesBlock'

import './Output.scss'
import BEM from '../../helpers/BEM'

const b = BEM('Output')

const Output = (props) => {
  const { title, output, allPurchases, error } = props

  if (allPurchases) {
    return (
      <div className={b()}>
        <h3 className={b('title')}>{title}</h3>
        {allPurchases.map(purchaseDate => (
          <PurchasesBlock key={purchaseDate.id} date={purchaseDate.id} allPurchases={purchaseDate.purchases}/>
        ))}
      </div>
    )
  }

  return (
    <div className={error ? b({ error }) : b()}>
      <h3 className={b('title')}>{title}</h3>
      <div className={b('message')}>{output ? output : props.children}</div>
    </div>
  )
}

Output.propTypes = {
  allPurchases: PropTypes.array,
  title: PropTypes.string,
  error: PropTypes.string,
  output: PropTypes.bool
}

export default Output