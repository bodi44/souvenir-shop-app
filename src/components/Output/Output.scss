.Output {
  position: relative;
  display: block;
  width: 100%;
  border-radius: 5px;
  background: lightcyan;
  margin: 10px auto;
  min-height: 200px;

  &_error {
    font-size: 16px;
    background: #ff7675;
  }

  &__title {
    padding-left: 20px;
    font-size: 20px;
  }

  &__message {
    padding-left: 20px;
  }
}