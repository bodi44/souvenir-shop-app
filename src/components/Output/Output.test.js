import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import expect from 'expect'
import Adapter from 'enzyme-adapter-react-16'

import Output from './Output'
import HelpMessage from '../HelpMessage/HelpMessage'

Enzyme.configure({ adapter: new Adapter() })

describe('Output Component', () => {
  it('should render correctly with title and output props', () => {

    const component = shallow(<Output title={'Output:'} output={'200 UAH'}/>)
    expect(component).toMatchSnapshot()
  })

  it('should render correctly with title, error props and given children', () => {
    const component = shallow(
      <Output title={'Output:'} error={'error'}>
        Some error occurred: <br/> To get instructions please type: <strong>help</strong>
      </Output>
    )
    expect(component).toMatchSnapshot()
  })

  it('should render correctly with title and children', () => {
    const component = shallow(
      <Output title={'Help:'}>
        <HelpMessage/>
      </Output>
    )
    expect(component).toMatchSnapshot()
  })

  it('should render correctly with title and allPurchases props', () => {
    const allPurchases = [
      {
        id: '2019-04-25',
        purchases: [
          {
            id: 'c4f63a46-de46-49c3-9773-0a47c62cdb77',
            date: '2019-04-25',
            price: 12,
            currency: 'EUR',
            productName: 'Photo'
          }
        ]
      }
    ]

    const component = shallow(
      <Output title={'Output:'} allPurchases={allPurchases}/>
    )
    expect(component).toMatchSnapshot()
  })
})