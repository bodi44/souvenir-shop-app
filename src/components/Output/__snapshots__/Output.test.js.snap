// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Output Component should render correctly with title and allPurchases props 1`] = `ShallowWrapper {}`;

exports[`Output Component should render correctly with title and children 1`] = `ShallowWrapper {}`;

exports[`Output Component should render correctly with title and output props 1`] = `ShallowWrapper {}`;

exports[`Output Component should render correctly with title, error props and given children 1`] = `ShallowWrapper {}`;
